frontend/src/components/resourceMap/details/KubeNodeDetails.tsx

```diff
@@ -0,0 +1,92 @@
+import { Box } from '@mui/system';
+import { memo } from 'react';
+import { KubeObject } from '../../../lib/k8s/cluster';
+import Deployment from '../../../lib/k8s/deployment';
+import Job from '../../../lib/k8s/job';
+import ReplicaSet from '../../../lib/k8s/replicaSet';
+import ConfigDetails from '../../configmap/Details';
+import CronJobDetails from '../../cronjob/Details';
+import DaemonSetDetails from '../../daemonset/Details';
+import EndpointDetails from '../../endpoints/Details';
+import HpaDetails from '../../horizontalPodAutoscaler/Details';
+import IngressClassDetails from '../../ingress/ClassDetails';
+import IngressDetails from '../../ingress/Details';
+import { LeaseDetails } from '../../lease/Details';
+import { LimitRangeDetails } from '../../limitRange/Details';
+import NamespaceDetails from '../../namespace/Details';
+import { NetworkPolicyDetails } from '../../networkpolicy/Details';
+import NodeDetails from '../../node/Details';
+import PodDetails from '../../pod/Details';
+import PDBDetails from '../../podDisruptionBudget/Details';
+import PriorityClassDetails from '../../priorityClass/Details';
+import ResourceQuotaDetails from '../../resourceQuota/Details';
+import RoleBindingDetails from '../../role/BindingDetails';
+import RoleDetails from '../../role/Details';
+import { RuntimeClassDetails } from '../../runtimeClass/Details';
+import SecretDetails from '../../secret/Details';
+import ServiceDetails from '../../service/Details';
+import ServiceAccountDetails from '../../serviceaccount/Details';
+import StatefulSetDetails from '../../statefulset/Details';
+import VolumeClaimDetails from '../../storage/ClaimDetails';
+import StorageClassDetails from '../../storage/ClassDetails';
+import VolumeDetails from '../../storage/VolumeDetails';
+import VpaDetails from '../../verticalPodAutoscaler/Details';
+import MutatingWebhookConfigList from '../../webhookconfiguration/MutatingWebhookConfigDetails';
+import ValidatingWebhookConfigurationDetails from '../../webhookconfiguration/ValidatingWebhookConfigDetails';
+import WorkloadDetails from '../../workload/Details';
+
+/**
+ * Shows details page for a given Kube resource
+ */
+export const KubeObjectDetails = memo(({ resource }: { resource: KubeObject }) => {
+  const kind = resource.kind;
+  const { name, namespace } = resource.metadata;
+
+  return (
+    <Box sx={{ overflow: 'hidden' }}>
+      <Box sx={{ marginTop: '-70px' }}>
+        {kind === 'Pod' && <PodDetails name={name} namespace={namespace} />}
```

Comment:
Seems like most of these take the name + namespace.
I wonder if we could just have a map like
```typescript
{
  Service: ServiceDetails,
  Pod: PodDetails,
  ...
}
```

I have also been thinking that maybe each KubeObject should have a list view and a details view components assigned to them, so we could extract this info. Maybe in a future PR.

Reply:
> I have also been thinking that maybe each KubeObject should have a list view and a details view components
yeah that'd be nice